tools/ci_connector_ops/ci_connector_ops/pipelines/bases.py

```diff
@@ -346,17 +404,27 @@ async def save_remote(self, local_path: Path, remote_key: str, content_type: str
             gcs_credentials=self.pipeline_context.ci_gcs_credentials_secret,
             flags=gcs_cp_flags,
         )
+        gcs_uri = "gs://" + self.pipeline_context.ci_report_bucket + "/" + remote_key
+        public_url = f"https://storage.googleapis.com/{self.pipeline_context.ci_report_bucket}/{remote_key}"
         if report_upload_exit_code != 0:
-            self.pipeline_context.logger.error(f"Uploading {local_path} to GCS Bucket: {self.pipeline_context.ci_report_bucket} failed.")
+            self.pipeline_context.logger.error(f"Uploading {local_path} to {gcs_uri} failed.")
+        else:
+            self.pipeline_context.logger.info(f"Uploading {local_path} to {gcs_uri} succeeded. Public URL: {public_url}")
         return report_upload_exit_code
 
     async def save(self) -> None:
         """Save the report files."""
         local_json_path = await self.save_local(self.json_report_file_name, self.to_json())
-        self.pipeline_context.logger.info(f"Report saved locally at {local_json_path}")
+        absolute_path = await local_json_path.absolute()
+        self.pipeline_context.logger.info(f"Report saved locally at {absolute_path}")
         if self.remote_storage_enabled:
             await self.save_remote(local_json_path, self.json_report_remote_storage_key, "application/json")
-            self.pipeline_context.logger.info(f"Report saved remotely at {self.json_report_remote_storage_key}")
+            for extra_file_path, extra_file_name in self.extra_files_to_upload:
+                await self.save_remote(extra_file_path, self.get_extra_file_remote_key(extra_file_name))
+
+    def get_extra_file_remote_key(self, filename: str) -> str:
```

Comment:
Nit: Maybe we rename this something generic like `compute_remote_storage_path` and reuse it in `json_report_remote_storage_key` as well